platform/osx/meson.build

```
@@ -0,0 +1,45 @@
+# The libraries and defines added in this file are the *requirements* that are
+# needed to build on the selected platform. Do not add debug/release/optional
+# compilation flags here.
+
+################################################################################
+# Commands to always run
+################################################################################
+_target_osx_logo = custom_target(
+                     'osx_logo',
+    output: ['logo.gen.h'],
+    input: ['logo.png'],
+    command: [SCRIPT_PLATFORM_LOGO, '@INPUT@', '@OUTPUT@', 'osx'],
+    build_by_default: true
+)
+SOURCES += [_target_osx_logo]
+
+
+################################################################################
+# Host Machine Specifics
+################################################################################
+if PLATFORM == 'osx'
+    MAIN_SOURCES += files('godot_main_osx.mm')
+
+    _osx_srcs = files('joypad_osx.cpp',
+                      #'context_gl_osx.mm',
+                      'crash_handler_osx.mm',
+                      'dir_access_osx.mm',
+                      'display_server_osx.mm',
+                      'godot_main_osx.mm',
+                      'os_osx.mm',
+                      'vulkan_context_osx.mm',
+                      )
+    CPP_ARGS += ['-DUNIX_ENABLED']
+    OBJCPP_ARGS += ['-DUNIX_ENABLED']
+    SOURCES += _osx_srcs
+    DEPENDENCIES += dependency('appleframeworks', modules: ['AppKit',
+                                                            'IOKit',
+                                                            'CoreAudio',
+                                                            'CoreServices',
+                                                            'Carbon',
+                                                            'GameController',
+                                                            'CoreHaptics',
+                                                            'ForceFeedback',
+                                                            'foundation'])
```

```suggestion
                                                            'Foundation',
                                                            'Cocoa',
                                                            'AudioUnit',
                                                            'CoreMIDI',
                                                            'CoreVideo',
                                                            'AVFoundation',
                                                            'CoreMedia',
                                                            'Metal',
                                                            'QuartzCore',
                                                            'IOSurface',
                                                            ])
```

And `MoltenVK.xcframework` from the Vulkan SDK, if static MoltenVK option is selected (which is the preferred way).